awx/main/models/credential/__init__.py

```diff
@@ -441,6 +458,8 @@ def get_input(self, field_name, **kwargs):
         :param field_name(str):        The name of the input field.
         :param default(optional[str]): A default return value to use.
         """
+        if field_name in self.dynamic_input_fields:
```

Comment:
For the sake of my dark paranoia, could you add to this condition:

```python
if self.kind != 'external' and field_name in self.dynamic_input_fields:
```

And then document that external credential sources _cannot_ ever be used to look up secrets for external credential sources. And everyone who had this thought should feel bad for thinking it.